### DISCUSSION TITLE
Terrain system, how big it can be?

### DISCUSSION THREAD

**Discussion by L4exp** [o3de/o3de#8672] [Help Wanted] [answered]
Maybe this question more theoretical than practical or i something misunderstanding, but still...
Terrain World component have some limitations per dimension(from -2048 to 2048) - 4096 m^2 can be rendered(let's exclude height).

In 'Terrain World Component's page wrote than limitation under 8 millon polygons.
Is there any way to awoid this limitation?
E.g. there are some generated maps by other software(~50+ km^2), which presented in tile format, and 4km^2 of them can be used in level.

In theory, terrain, which presented in tiles format(quad shape + height gradient e.g. by 1024m per tile and count of them can be high)
can be loaded partially - only nearest neibourhood tiles(we need to render - which we can see).

Maybe now existing ways to do this,
maybe in future it can be integrated,
or there are no plans to do this?

Thanks in advance!

**Comment by soltsmank** [comment]
Found some useful info:
[A Cloud-based Large Terrain System for O3DE Gems - Chen Wei, Huawei](https://youtu.be/xbhGbSbKpvM)
[Cesium for O3DE](https://cesium.com/platform/cesium-for-o3de/)

**Comment by L4exp** [comment, marked as answer]
Seems like Terrain Experemental Gem got some improvements. Terrain gem getting multy mesh visualizing system
